packages/nextjs/src/config/templates/routeHandlerWrapperTemplate.ts

```diff
@@ -2,12 +2,29 @@ import * as Sentry from '@sentry/nextjs';
 import type { WebFetchHeaders } from '@sentry/types';
 // @ts-expect-error Because we cannot be sure if the RequestAsyncStorage module exists (it is not part of the Next.js public
 // API) we use a shim if it doesn't exist. The logic for this is in the wrapping loader.
-import { requestAsyncStorage } from '__SENTRY_NEXTJS_REQUEST_ASYNC_STORAGE_SHIM__';
+import * as origModule from '__SENTRY_NEXTJS_REQUEST_ASYNC_STORAGE_SHIM__';
 // @ts-expect-error See above
 import * as routeModule from '__SENTRY_WRAPPING_TARGET_FILE__';
 
 import type { RequestAsyncStorage } from './requestAsyncStorageShim';
 
+type NextAsyncStorageModule =
+  | {
+      workUnitAsyncStorage: RequestAsyncStorage;
+    }
+  | {
+      requestAsyncStorage: RequestAsyncStorage;
+    };
+
+const asyncStorageModule = { ...origModule } as NextAsyncStorageModule;
```

Comment:
Did you find a reason that we should clone the object? It may be a bit expensive to do so in various ways.

Reply:
Unfortunately, `requestAsyncStorage is not exported` is returned when not use clone.